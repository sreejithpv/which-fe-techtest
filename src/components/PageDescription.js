import React from 'react';
import PropTypes from 'prop-types';

const PageDescription = ({ children  }) => (
  <p data-testid="page-description">{ children }</p>
);

PageDescription.propTypes = {
  children: PropTypes.node,
};

PageDescription.defaultProps = {
  children: null,
}

export default PageDescription;