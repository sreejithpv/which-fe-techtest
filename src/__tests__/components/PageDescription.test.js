import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import PageDescription from '../../components/PageDescription';

describe('PageDescription Component', () => {
  const { getByTestId } = render(<PageDescription>This is page description</PageDescription>);

  it('displays the give PageDescription content', () => {
    const tree = getByTestId('page-description');
    expect(tree.innerHTML).toEqual('This is page description');
  });

  test("Render snapshot the component", () => {
    const tree = TestRenderer.create(<PageDescription>Snapshot PageDescription</PageDescription>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});