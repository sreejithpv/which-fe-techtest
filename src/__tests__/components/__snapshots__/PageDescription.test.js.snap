// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageDescription Component Render snapshot the component 1`] = `
<p
  data-testid="page-description"
>
  Snapshot PageDescription
</p>
`;
